### PR TITLE
[gs] lstrip the url_tup.path before using it

### PR DESCRIPTION
**Pull request by fujin**
* strips the leading /

note: I tested this on top of v1.0 (.3) since that is what is available in pypi.

### CODE REVIEW

File: wal_e/blobstore/gs/utils.py [hunk 1]
```diff
@@ -24,10 +24,11 @@ def _uri_to_blob(creds, uri, conn=None):
     assert uri.startswith('gs://')
     url_tup = urlparse(uri)
     bucket_name = url_tup.netloc
+    lstrip_path = url_tup.path.lstrip('/')
     if conn is None:
         conn = calling_format.connect(creds)
     b = storage.Bucket(conn, name=bucket_name)
-    return storage.Blob(url_tup.path, b)
+    return storage.Blob(lstrip_path, b)
```
Comment (fujin) [inline wal_e/blobstore/gs/utils.py]: could maybe do the lstrip here

**Comment by fdr** [comment]
This does break old backups, so we're going to have to do something about that, e.g. some kind of backwards compatibility code. Also, have you tested to make sure this handles all of backup fetch/push and wal fetch/push consistently?

**Comment by erikgrinaker** [comment]
`backup-list` does not work with this change. Occurances of `prefix='/' + prefix` in `worker/gs/gs_worker.py` must be replaced with `prefix=prefix`.

**Comment by erikgrinaker** [comment]
`backup-fetch` was similarly broken, replace `'/' + part_abs_name` with `part_abs_name` in `worker/gs/gs_worker.py`.

**Comment by fdr** [comment]
Can someone submit and test an updated patch? Also see my comment in #308: https://github.com/wal-e/wal-e/pull/342#discussion_r122843766. If we are going to break people's configurations in 1.1, at least they need to be able to access old backups if they insert an extra slash.

**Comment by fujin** [comment]
Fixed both of the things @erikgrinaker mentioned; still need to come up with a way to preserve backwards compatibility though?

**Comment by deverant** [comment]
Anybody have any good ideas how to handle the backwards compatibility? I wouldn't mind getting this issue fixed.

Would we be fine with some `WALE_GS_ADD_EXTRA_SLASH` environment variable that could only be used when restoring a backup? Not really sure we would be fine allowing people to keep uploading with a weird extra slash in the path..

**Comment by fdr** [comment]
On Fri, Oct 27, 2017 at 1:27 PM Samuel Kohonen <notifications@github.com>
wrote:

> Anybody have any good ideas how to handle the backwards compatibility? I
> wouldn't mind getting this issue fixed.
>
> Would we be fine with some WALE_GS_ADD_EXTRA_SLASH environment variable
> that could only be used when restoring a backup? Not really sure we would
> be fine allowing people to keep uploading with a weird extra slash in the
> path..
>

I think that restore env var would be fine. Note that Swift also has such a
bug, so I think it could be more simply called WALE_ADD_EXTRA_SLASH.

**Pull request closed without merging**